### PR TITLE
Create a github action that compiles the pdf

### PR DESCRIPTION
**Pull request by badetitou**
These modifications add Github actions that compile and publish as release the last version of sample.tex as a file named "cv.pdf".
It is thus possible to get the last version of the generated pdf at the URL: https://github.com/:owner/AltaCV/releases/download/latest/cv.pdf

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the main branch
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  schedule:
+  - cron: "0 12 1 * *"
```
Comment (rstm-sf) [inline .github/workflows/main.yml:13]: Hello!

Why is this needed?

Comment (badetitou) [reply .github/workflows/main.yml:13]: Hmmm, in fact. It might be useless. 
I did not GitHub actions when I have tried all of this.

We should remove this. I'm doing it

**Comment by badetitou** [comment]
So, 

I have detected a bug in my fork, and tried to understand where is the problem.
It appears that my first PR was really bad (bibtex action does the latex+bibtex+latex+latex), and the latex actions were bad.

I have updated the latex action, so now it works, and it is faster (you can have a look at my fork in the _**main**_ branch.
I also have updated the _.cls_ file because there is a bug with bib ion the last version. I only imply the quick fix proposed here: https://tex.stackexchange.com/questions/564990/error-after-miktex-reinstall-text-line-contains-an-invalid-character

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Build pdf
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the main branch
+on:
+  push:
+    branches: [ main ]
+
```
Comment (rstm-sf) [inline .github/workflows/main.yml:10]: It would be nice to add an option to the `pull_request` as well, you can also say that the `push` option is triggered on file changes in a specific path or by extension
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+          # LaTeX engine to be used
+          # compiler: pdflatex # optional, default is latexmk
+          # Extra arguments to be passed to the latex compiler
+          # args: ""
+          # Install extra packages by apt-get
+          # extra_system_packages: # optional
+
+                        
+      - name: Rename pdf
+        run: mv sample.pdf cv.pdf
```
Comment (rstm-sf) [inline .github/workflows/main.yml:46]: This can be set using the `-jobname=cv` option to `args` by compile latex

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      
+      - name: Create empty changelog
```
Comment (rstm-sf) [inline .github/workflows/main.yml:48]: Sorry, but I do not understand why this is needed. Could you please clarify why this is needed?

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        run: echo "# Update CV" > CHANGELOG.md
+        
+      - name: Update Release
```
Comment (rstm-sf) [inline .github/workflows/main.yml:51]: It seems to me that it is better to separate this part so that it runs only in the master branch, for example, by condition
`if: github.ref == 'refs/heads/master'`
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif

Also, it would be nice to use `actions/upload-artifact` and `actions/download-artifact`, so that we can see which file is still on the pull request, and in the wizard, download the artifacts and adding in to release

To add assets to the release, we can also use `actions/upload-release-asset` or `svenstaro/upload-release-action`

File: altacv.cls [hunk 6]
```diff
@@ -38,6 +38,7 @@
 %% v1.3.2 Hopefully this helps make the PDF
 %% file more 'friendly' with copy-paste etc
 \RequirePackage[a-1b]{pdfx}
+\catcode30=12
```
Comment (rstm-sf) [inline altacv.cls:41]: Could you please clarify why this is in this PR and not separate? It would be nice to configure only CI/CD in these changes

**Review by rstm-sf — COMMENTED** [review]
Hi, I did a [little research](https://github.com/rstm-sf/cv/blob/2859cc0ac5f13d9afa57e78aa2da9adc7089e077/.github/workflows/latex.yml#L1) on configuring up CI/CD with github actions. I would like to ask a few questions and suggest changes

**Comment by liantze** [comment]
Hello @badetitou, thanks for the contributions and discussions! Personally I would prefer this repo (i.e. my own) to contain LaTeX-only sources. I would strongly encourage that you host/fork these at your own repo.

**Comment by liantze** [comment]
@badetitou Please feel free to submit a separate PR for the pdfx/biblatex incompatibility if you wish; or I can update altacv.cls sometime soon.

**Comment by rstm-sf** [comment]
@liantze these changes help to determine early on that the submitted changes have broken something. So I don't understand why you give them up

**Comment by liantze** [comment]
@rstm-sf Thanks again for the discussion, and sorry about this. I will admit that it was purely due to my own preference that my own copy (the original anyway) of this repo contains only LaTeX-related code. Github actions, docker customisations are great extras! and I'm happy to see forks and customisations on these aspects by other users.

I suppose it is my own fault that I did not comment earlier on this PR (and other similar PRs) due to various other time-shortness/commitments in real life to react quickly on my own repos... I guess I should probably close/freeze my repos, at least until I have more time and energy on my hands.

**Comment by rstm-sf** [comment]
Ok, thanks for the answer ;)

File: .github/workflows/main.yml (hunk 3 above)
Comment (badetitou) [reply .github/workflows/main.yml:46]: Oo. Ok, that's nice ! (I'll not modify it since it works for my case, for my CV)
But cool!

File: .github/workflows/main.yml (hunk 4 above)
Comment (badetitou) [reply .github/workflows/main.yml:48]: It is not needed, but the point is, when people download my cv, I prefer that they see "cv.pdf" than "sample.pdf"

File: .github/workflows/main.yml (hunk 5 above)
Comment (badetitou) [reply .github/workflows/main.yml:51]: OOO
NICE! 

I definitely believe this is the good way to follow.
I should spend time on it!!!

File: altacv.cls (hunk 6 above)
Comment (badetitou) [reply altacv.cls:41]: It is needed to compile using the docker image used by the github action. As a fact, we can make another PR.
But this bug will also be fixed in future version xxx (I do not remember that latex package that fails)

**Comment by badetitou** [comment]
@liantze 
No problem, I'm already using the github action on my own repo (and use the generated link to the pdf in my personal website :) )

Thanks for your incredible work on this CV template!!

@rstm-sf 
Thanks for all your comments. I should spend more time on GitHub actions. I believe we can create super cool automated stuff.
